vra/resource_network_ip_range.go

```diff
@@ -0,0 +1,196 @@
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceNetworkIPRange() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNetworkIPRangeCreate,
+		Read:   resourceNetworkIPRangeRead,
+		Update: resourceNetworkIPRangeUpdate,
+		Delete: resourceNetworkIPRangeDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"fabric_network_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"start_ip_address": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				// Do we need to validate?
+			},
+			"end_ip_address": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				// Do we need to validate?
+			},
+			"ip_version": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"IPv4", "IPv6"}, true),
+			},
+			"progress": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"created_at": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"external_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"org_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tags":  tagsSchema(),
+			"links": linksSchema(),
+		},
+	}
+}
+
+func resourceNetworkIPRangeCreate(d *schema.ResourceData, m interface{}) error {
+	log.Printf("Starting to create vra_network_profile resource")
+	apiClient := m.(*Client).apiClient
+
+	name := d.Get("name").(string)
+	endIPAddress := d.Get("end_ip_address").(string)
+	startIPAddress := d.Get("start_ip_address").(string)
+
+	networkIPRangeSpecification := models.NetworkIPRangeSpecification{
+		EndIPAddress:    &endIPAddress,
+		FabricNetworkID: d.Get("fabric_network_id").(string),
+		IPVersion:       d.Get("ip_version").(string),
+		Name:            &name,
+		StartIPAddress:  &startIPAddress,
+		Tags:            expandTags(d.Get("tags").(*schema.Set).List()),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		networkIPRangeSpecification.Description = v.(string)
+	}
+
+	log.Printf("[DEBUG] create network ip rangee: %#v", networkIPRangeSpecification)
+
+	createNetworkIPRangeCreated, createNetworkIPRangeAccepted, err := apiClient.NetworkIPRange.CreateInternalNetworkIPRange(network_ip_range.NewCreateInternalNetworkIPRangeParams().WithBody(&networkIPRangeSpecification))
+	if err != nil {
+		return err
+	}
+	if createNetworkIPRangeCreated != nil {
+		d.SetId(*createNetworkIPRangeCreated.Payload.ID)
+	} else {
+		d.SetId(*createNetworkIPRangeAccepted.Payload.ID)
+	}
```

Comment:
Payload returned by [`createNetworkIPRangeCreated`](https://github.com/vmware/vra-sdk-go/blob/master/pkg/client/network_ip_range/create_internal_network_ip_range_responses.go#L74) and [`createNetworkIPRangeAccepted`](https://github.com/vmware/vra-sdk-go/blob/master/pkg/client/network_ip_range/create_internal_network_ip_range_responses.go#L107) is of type RequestTracker.

You may refer to [`vra_machine`](https://github.com/vmware/terraform-provider-vra/blob/master/vra/resource_machine.go#L248) or other IaaS resources for an example.

Reply:
@dmettem  - I agree the swagger says it's supposed to return a response of type RequestTracker, but I'm not seeing that in practice at least not on-prem ....I don't have access to vRA Cloud to validate if it behaves differently or not, so wondering if someone else can. I've uploaded the relevant snippet from a log to https://gist.github.com/cars/83b6b4955a2400927f02a09676607da6#file-tf_network_ip_range-log 

The POST starts  [here](https://gist.github.com/cars/83b6b4955a2400927f02a09676607da6#file-tf_network_ip_range-log-L21)  And the non-RequestTracker response is [here](https://gist.github.com/cars/83b6b4955a2400927f02a09676607da6#file-tf_network_ip_range-log-L52)

I wondering if maybe there's an error in the swagger or ... ?

Reply:
@cars - Sorry for late response. I'll verify this on 8.1 build and get back.

Reply:
@cars - From the snippet you provided, I see that you are not getting the RequestTracker in the response. I opened an internal issue to correct the swagger. 

For now, you can ignore this comment, and work on the rest to merge into master. Thank You!